docs/book/resource_layer.rst

```diff
@@ -0,0 +1,187 @@
+
+The repository service is available via the *sylius.repository.product* id and can be used like all the repositories you have seen before.
+
+.. code-block:: php
+
+    <?php
+
+    public function myAction()
+    {
+        $repository = $this->container->get('sylius.repository.product');
+
+        $product = $repository->find(4); // Get product with id 4, returns null if not found.
+        $product = $repository->findOneBy(['slug' => 'my-super-product']); // Get one product by defined criteria.
+
+        $products = $repository->findAll(); // Load all the products!
+        $products = $repository->findBy(['special' => true]); // Find products matching some custom criteria.
+    }
+
+Every Sylius repository supports paginating resources. To create a `Pagerfanta instance <https://github.com/whiteoctober/Pagerfanta>`_ use the ``createPaginator`` method.
+
+.. code-block:: php
+
+    <?php
+
+    public function myAction(Request $request)
+    {
+        $repository = $this->container->get('sylius.repository.product');
+
+        $products = $repository->createPaginator();
+        $products->setMaxPerPage(3);
+        $products->setCurrentPage($request->query->get('page', 1));
+
+        // Now you can return products to template and iterate over it to get products from current page.
+    }
+
+Paginator can be created for a specific criteria and with desired sorting.
+
+.. code-block:: php
+
+    <?php
+
+    public function myAction(Request $request)
+    {
+        $repository = $this->container->get('sylius.repository.product');
+
+        $products = $repository->createPaginator(['foo' => true], ['createdAt' => 'desc']);
+        $products->setMaxPerPage(3);
+        $products->setCurrentPage($request->query->get('page', 1));
+    }
+
+Controller
+~~~~~~~~~~
+
+This service is the most important for every resource and provides a format agnostic CRUD controller with the following actions:
+
+* [GET]      showAction() for getting a single resource
+* [GET]      indexAction() for retrieving a collection of resources
+* [GET/POST] createAction() for creating new resource
+* [GET/PUT]  updateAction() for updating an existing resource
+* [DELETE]   deleteAction() for removing an existing resource
+
+As you see, these actions match the common operations in any REST API and yes, they are format agnostic.
+This means, all Sylius controllers can serve HTML, JSON or XML, depending on what you request.
+
+Additionally, all these actions are very flexible and allow you to use different templates, forms, repository methods per route.
+The bundle is very powerful and allows you to register your own resources as well. To give you some idea of what is possible, here are some examples!
+
+Displaying a resource with a custom template and repository methods:
+
+.. code-block:: yaml
+
+    # routing.yml
+
```

Comment:
The symfony docs would not have a line break here: http://symfony.com/doc/current/book/controller.html#mapping-a-url-to-a-controller